### PR TITLE
Support parsing SQL Server SELECT name, sql

### LINKED ISSUE #29174: Support parsing SQL Server SELECT name, sql
# Background
Hi community.
This issue is for #29149.

The ShardingSphere SQL parser engine helps users to parse SQL to create the AST (Abstract Syntax Tree) and visit the AST to get SQLStatement (Java Object). Currently, we are planning to enhance the support for `SQL Server` SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

# Task
This issue is to support more `SQL Server` sql parsing, as follows:
```sql
SELECT name, server_id, provider
FROM sys.servers
```
[link](https://learn.microsoft.com/zh-cn/sql/database-engine/configure-windows/view-or-change-server-properties-sql-server?view=sql-server-ver16)

```sql
SELECT ProductId, Name, value
FROM Product
    CROSS APPLY STRING_SPLIT(Tags, ',')
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/functions/string-split-transact-sql?view=azure-sqldw-latest)

```sql
SELECT value as tag, COUNT(*) AS [number_of_articles]
FROM Product
    CROSS APPLY STRING_SPLIT(Tags, ',')
GROUP BY value
HAVING COUNT(*) > 2
ORDER BY COUNT(*) DESC
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/functions/string-split-transact-sql?view=azure-sqldw-latest)

```sql
SELECT ProductId, Name, Tags
FROM Product
WHERE EXISTS (SELECT *
    FROM STRING_SPLIT(Tags, ',')
    WHERE value IN ('clothing', 'road'))
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/functions/string-split-transact-sql?view=azure-sqldw-latest)

```sql
INSERT INTO iris_rx_data ("Sepal.Length", "Sepal.Width", "Petal.Length", "Petal.Width" , "Species")
EXECUTE sp_execute_external_script
  @language = N'R'
  , @script = N'iris_data <- iris
```
[link](https://learn.microsoft.com/zh-cn/sql/machine-learning/predictions/native-scoring-predict-transact-sql?view=azure-sqldw-latest)

# Process
1. First confirm that this is a correct `SQL Server` sql syntax, if not please leave a message under the issue and ignore it;
2. Compare SQL definitions in Official SQL Doc and ShardingSphere SQL Doc;
3. If there is any difference in ShardingSphere SQL Doc, please correct them by referring to the Official SQL Doc;
4. Run mvn install the current_file_module;
5. Check whether there are any exceptions. If indeed, please fix them. (Especially xxxVisitor.class);
6. Add new corresponding SQL case in SQL Cases and expected parsed result in Expected Statement XML;
7. Run SQLParserParameterizedTest to make sure no exceptions.

# Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr `g4` file
3. Be familiar with `SQL Server` SQLs

### PR DESCRIPTION
**Pull request by yydeng626**
Fixes #29174.

Changes proposed in this pull request:
  - add execute clause segment [offical link](https://learn.microsoft.com/en-us/sql/t-sql/language-elements/execute-transact-sql?view=sql-server-ver16)

```
SELECT ProductId, Name, value
FROM Product
    CROSS APPLY STRING_SPLIT(Tags, ',')
```

```
SELECT value as tag, COUNT(*) AS [number_of_articles]
FROM Product
    CROSS APPLY STRING_SPLIT(Tags, ',')
GROUP BY value
HAVING COUNT(*) > 2
ORDER BY COUNT(*) DESC
```

```
SELECT ProductId, Name, Tags
FROM Product
WHERE EXISTS (SELECT *
    FROM STRING_SPLIT(Tags, ',')
    WHERE value IN ('clothing', 'road'))
```

```
INSERT INTO iris_rx_data ("Sepal.Length", "Sepal.Width", "Petal.Length", "Petal.Width" , "Species")
EXECUTE sp_execute_external_script
  @language = N'R'
  , @script = N'iris_data <- iris'
```
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by strongduanmu** [comment]
Hi @yydeng626, can you solve code conflict?

**Comment by yydeng626** [comment]
@strongduanmu Please check again

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/sqlserver/dml/SQLServerCallStatement.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.apache.shardingsphere.sql.parser.sql.common.segment.dml.expr.ExpressionSegment;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dml.CallStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.sqlserver.SQLServerStatement;
+
+import java.util.List;
+
+/**
+ * MySQL call statement.
```
Comment (strongduanmu) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/sqlserver/dml/SQLServerCallStatement.java:30]: Please replace MySQL with SQLServer.

File: test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/internal/asserts/segment/insert/InsertExecClauseAssert.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.shardingsphere.test.it.sql.parser.internal.asserts.segment.owner.OwnerAssert;
+import org.apache.shardingsphere.test.it.sql.parser.internal.cases.parser.jaxb.segment.impl.exec.ExpectedExecClause;
+import org.hamcrest.CoreMatchers;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * insert execute clause assert.
```
Comment (strongduanmu) [inline test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/internal/asserts/segment/insert/InsertExecClauseAssert.java:36]: Please use Insert for first word.

File: test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/internal/asserts/segment/insert/InsertExecClauseAssert.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ **/
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class InsertExecClauseAssert {
+    
+    /**
+     *  Assert actual execute segment is correct with expected execute clause.
```
Comment (strongduanmu) [inline test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/internal/asserts/segment/insert/InsertExecClauseAssert.java:42]: Please add new line after java doc.

File: test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/internal/asserts/segment/insert/InsertExecClauseAssert.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    public static void assertIs(final SQLCaseAssertContext assertContext, final ExecSegment actual, final ExpectedExecClause expected) {
+        assertThat(assertContext.getText("exec procedure name assertion error: "), actual.getProcedureName().getIdentifier().getValue(), CoreMatchers.is(expected.getName()));
+        if (null == expected.getOwner()) {
+            assertFalse(actual.getProcedureName().getOwner().isPresent(), assertContext.getText("Actual owner should not exist."));
+        } else {
+            assertTrue(actual.getProcedureName().getOwner().isPresent(), assertContext.getText("Actual owner should exist."));
+            OwnerAssert.assertIs(assertContext, actual.getProcedureName().getOwner().get(), expected.getOwner());
+        }
+
```
Comment (strongduanmu) [inline test/it/parser/src/main/java/org/apache/shardingsphere/test/it/sql/parser/internal/asserts/segment/insert/InsertExecClauseAssert.java:55]: Please remove this useless blank line.

File: test/it/parser/src/main/resources/case/dml/select-special-function.xml [hunk 5]
```diff
@@ -591,4 +591,44 @@
             </subquery-table>
         </from>
     </select>
+
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select-special-function.xml:594]: Please keep one blank line here.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
@yydeng626 Good job, merged.

**Pull request merged**